### PR TITLE
Card ingestion test

### PR DESCRIPTION
**Pull request by qnkhuat**
tests for card ingestion. More to come
Ref #50261

### CODE REVIEW

File: src/metabase/models/card.clj [hunk 1]
```diff
@@ -1053,13 +1053,13 @@
...
                                                         [:= :mr.moderated_item_id :this.id]
-                                                        [:= :mr.most_recent true]]]
+                                                        [:= :mr.most_recent true]]]}})
                   ;; Workaround for dataflow :((((((
                   ;; NOTE: disabled for now, as this is not a very important ranker and can afford to have stale data,
                   ;;       and could cause a large increase in the query count for dashboard updates.
                   ;;       (see the test failures when this hook is added back)
                   ;:dashcard  [:model/DashboardCard [:= :dashcard.card_id :this.id]]
-                  }})
+
```
Comment (crisptrutski) [inline src/metabase/models/card.clj:1062]: This commented out code belongs inside the map. Did you move this intentionally or was it VIM?

File: test/metabase/search/postgres/index_test.clj [hunk 2]
```diff
@@ -204,3 +205,37 @@
...
+  (t2/query {:select [:*]
+             :from   [search.index/*active-table*]
+             :where  [:and
+                      [:= :name entity-name]
+                      [:= :model model]]}))
+
+(deftest card-ingestion-test
+  (search.tu/with-temp-index-table
+    (testing "dashboard card count"
+      (let [search-id (mt/random-name)]
```
Comment (crisptrutski) [inline test/metabase/search/postgres/index_test.clj:226]: Sounds like a db id. How about also calling this `entity-name`?

File: src/metabase/models/card.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/models/card.clj:1062]: vim :sad:
do that everytime I open the file. will revert

File: src/metabase/search/postgres/index.clj [hunk 3]
```diff
@@ -49,38 +49,38 @@
...
-        ;; permission related entities
+         ;; permission related entities
          [:collection_id :int]
          [:database_id :int]
-        ;; filter related
+         ;; filter related
          [:archived :boolean :not-null [:default false]]
          [:creator_id :int]
-         [:last_edited_at :timestamp]
+         [:last_edited_at :timestamp-with-time-zone]
```
Comment (qnkhuat) [inline src/metabase/search/postgres/index.clj:75]: should all be with time zone
ref: https://github.com/metabase/metabase/issues/33577

File: test/metabase/search/fulltext/scoring_test.clj [hunk 4]
```diff
@@ -2,91 +2,47 @@
...
 (set! *warn-on-reflection* true)
 
-(def ^:private ^:dynamic *user-ctx* nil)
-
 ;; We act on a random localized table, making this thread-safe.
 #_{:clj-kondo/ignore [:metabase/test-helpers-use-non-thread-safe-functions]}
 (defmacro with-index-contents
   "Populate the index with the given appdb agnostic entity shapes."
   [entities & body]
-  `(when (search/supports-index?)
```
Comment (qnkhuat) [inline test/metabase/search/fulltext/scoring_test.clj:29]: moved to metabase.search.test_util

File: src/metabase/models/card.clj [hunk 5]
```diff
@@ -1053,13 +1053,13 @@
...
                                                         [:= :mr.moderated_item_id :this.id]
-                                                        [:= :mr.most_recent true]]]
+                                                        [:= :mr.most_recent true]]]}
                   ;; Workaround for dataflow :((((((
                   ;; NOTE: disabled for now, as this is not a very important ranker and can afford to have stale data,
                   ;;       and could cause a large increase in the query count for dashboard updates.
                   ;;       (see the test failures when this hook is added back)
                   ;:dashcard  [:model/DashboardCard [:= :dashcard.card_id :this.id]]
-                  }})
+   #_:end})
```
Comment (crisptrutski) [inline src/metabase/models/card.clj:1062]: Sneaky solution 🧠

File: test/metabase/search/postgres/index_test.clj [hunk 6]
```diff
@@ -204,3 +205,98 @@
...
+              yesterday    (t/- (t/offset-date-time) (t/days 1))
+              two-days-ago (t/- yesterday (t/days 1))]
+          (mt/with-temp
+            [:model/Collection    {coll-id :id}      {:name            "My collection"
+                                                      ;; :official_collection = true
+                                                      :authority_level "official"}
+             :model/Card          {card-id :id}      {:name                search-term
+                                                      :type                card-type
+                                                      :query_type          "query"
+                                                      ;; :database_id = (mt/id)
```
Comment (crisptrutski) [inline test/metabase/search/postgres/index_test.clj:254]: This comment doesn't seem necessary

Comment (crisptrutski) [reply test/metabase/search/postgres/index_test.clj:254]: I would just stick to the counter intuitive ones 🤷 

Even the created_at => model_created_at stuff is obvious if you just look at how the variables are used.

**Review by crisptrutski — APPROVED** [review]
Awesome!

File: test/metabase/search/postgres/index_test.clj (hunk 6 above)
Comment (qnkhuat) [reply test/metabase/search/postgres/index_test.clj:254]: I just want to keep the convention so we know where stuffs come from, this one is obvious but the pined, last_edited_at stuff is not.

**Pull request merged**